docs/docusaurus/docs/core/connect_to_data/dataframes/dataframes.md

```diff
@@ -1,9 +1,198 @@
+- <PrereqDataContext/>.  These examples assume the variable `context` contains your Data Context.
+
+<Tabs>
+
+<TabItem value="procedure" label="Procedure">
+
+1. Define the Data Source parameters.
+
+   A dataframe Data Source requires the following information:
+
+   - `name`: A name by which to reference the Data Source.  This should be unique among all Data Sources on the Data Context.
+
+   Update `data_source_name` in the following code with a descriptive name for your Data Source:
+
+   ```python title="Python" name="docs/docusaurus/docs/core/connect_to_data/dataframes/_examples/_spark_df_data_source.py - define Data Source parameters"
+   ```
+
+2. Create the Data Source.
+
+   To read a pandas dataframe you will need to create a pandas Data Source.  Likewise, to read a Spark dataframe you will need to create a Spark Data Source.
+
+   <Tabs queryString="execution_engine" groupId="execution_engine" defaultValue='pandas'>
+
+      <TabItem value="pandas" label="pandas">
+
+      Execute the following code to create a pandas Data Source:
+
+      ```Python title="Python" name="docs/docusaurus/docs/core/connect_to_data/dataframes/_examples/_pandas_df_data_source.py - create Data Source"
+      ```
+
+      </TabItem>
+
+      <TabItem value="spark" label="Spark">
+
+      Execute the following code to create a Spark Data Source:
+
+      ```Python title="Python" name="docs/docusaurus/docs/core/connect_to_data/dataframes/_examples/_spark_df_data_source.py - create Data Source"
+      ```
+
+      </TabItem>
+
+   </Tabs>
+
+</TabItem>
+
+<TabItem value="sample_code" label="Sample code">
+
+   <Tabs queryString="execution_engine" groupId="execution_engine" defaultValue='pandas'>
+
+      <TabItem value="pandas" label="pandas">
+
+      ```Python title="Python" name="docs/docusaurus/docs/core/connect_to_data/dataframes/_examples/_pandas_df_data_source.py - full example"
+      ```
+
+      </TabItem>
+
+      <TabItem value="spark" label="Spark">
+
+      ```Python title="Python" name="docs/docusaurus/docs/core/connect_to_data/dataframes/_examples/_spark_df_data_source.py - full example"
+      ```
+
+      </TabItem>
+
+   </Tabs>
+
+</TabItem>
+
+</Tabs>
+
+## Create a Data Asset and Batch Definition
+
+To access data from your dataframe in GX you will connect to the dataframe with a Data Asset.  Then you will define a Batch Definition with which the data can be retrieved.
+
+Because dataframes exist in memory and cease to exist when the Python session ends a dataframe Data Asset will need to be created anew in every Python session that utilizes it.
+
+In a File Data Context previous dataframe Data Asset and Batch Definition configurations will persist between sessions.  However, since the dataframe they connected to will not also persist between Python sessions those configurations will only be useable for reference purposes.  
```

Comment:
Hopefully this will change with https://greatexpectations.atlassian.net/browse/V1-384